lib/nvme/nvme_pcie.c

```diff
@@ -1841,7 +1841,24 @@ nvme_pcie_qpair_check_timeout(struct spdk_nvme_qpair *qpair)
 		 * Request has timed out. This could be i/o or admin request.
 		 * Call the registered timeout function for user to take action.
 		 */
-		ctrlr->timeout_cb_fn(ctrlr->timeout_cb_arg, ctrlr, qpair, tr->cid);
+
```

Comment:
The logic in this function was fine when the user's only recourse for any time out was to do a full controller reset, but if we're going to allow users to send ABORTs for specific commands, this function needs to loop over the entire list of trackers and call the callback for each one that is timed out. Right now, it only calls the callback for the oldest one.